File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/infrastructure/resolution/FieldResolver.java
package com.gonezo.multiplatform.audioextraction.infrastructure.resolution;

import com.gonezo.multiplatform.audioextraction.domain.model.FieldCandidate;
import com.gonezo.multiplatform.audioextraction.domain.model.ResolvedField;
import com.gonezo.multiplatform.audioextraction.domain.schema.FieldSchema;
import java.util.List;

interface FieldResolver {
  boolean supports(FieldSchema schema);

  ResolvedField resolve(String fieldName, FieldSchema schema, List<FieldCandidate> candidates);
}

File: core/bin/main/com/gonezo/audioextraction/infrastructure/resolution/FieldResolver.kt
package com.gonezo.audioextraction.infrastructure.resolution

import com.gonezo.audioextraction.domain.model.FieldCandidate
import com.gonezo.audioextraction.domain.model.ResolvedField
import com.gonezo.audioextraction.domain.schema.FieldSchema

interface FieldResolver {
  fun supports(schema: FieldSchema): Boolean
  fun resolve(fieldName: String, schema: FieldSchema, candidates: List<FieldCandidate>): ResolvedField
}
